components/nexus-repository/src/main/java/org/sonatype/nexus/repository/rest/api/RepositoriesApiResource.java

```diff
@@ -83,6 +83,19 @@ public Response deleteRepository(@PathParam("repositoryName") final String repos
     return Response.status(isDeleted ? NO_CONTENT : NOT_FOUND).build();
   }
 
+  @Override
+  @GET
+  @Path("/{repositoryName}")
+  @RequiresAuthentication
+  public AbstractApiRepository getRepository(@PathParam("repositoryName") final String repositoryName) {
+    try {
+      return convert(authorizingRepositoryManager.getEditableRepositoryOrThrow(repositoryName));
```

Comment:
Method `getEditableRepositoryOrThrow` checks for administrative `edit` permission (it calls `ensureUserCanAdmin(EDIT, `) before it returns and this isn't what you should be testing for here. You should check the `read` permission instead (`userHasRepositoryViewPermissionTo(READ, )`).  Please consider adding some tests for the new endpoint in `org.sonatype.nexus.testsuite.rest.RepositoriesApiResourceGetMember` that would cover permission checks.

Reply:
I've updated the permission check to match what list() is doing. I can't get the unit tests in master to pass on Windows so I'll look into running in a VM later.